pkgs/tools/typesetting/stapler/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildPythonApplication, fetchFromGitHub, pypdf2, more-itertools }:
+
+buildPythonApplication rec {
+  pname = "stapler";
+  version = "2016-04-24";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "hellerbarde";
+    repo = pname;
+    rev = "7c153e6a8f52573ff886ebf0786b64e17699443a";
+    sha256 = "1wyrsfp2zrc23k853if4lp2lp0rzscsg1g4wijks0gha1zap94wy";
+  };
+
+  propagatedBuildInputs = [ pypdf2 more-itertools ];
+
+  checkPhase = ''
+    staplelib/tests.py
```

Comment:
Python is not in PATH here so need to do `patchShebangs staplelib/tests.py` before running the test script.

Reply:
Actually also need to `patchShebangs stapler` since the test script runs the stapler command. I'd suggest to have something like
```
  patchPhase = ''
    patchShebangs staplelib/tests.py
    patchShebangs stapler
  '';

  checkPhase = "staplelib/tests.py";
```